rfcs/0004-fulltext-search.md

```diff
@@ -0,0 +1,223 @@
+# RFC-0000: Template
+
+<dl>
+  <dt>Author</dt>
+  <dd>Ford Nickels</dd>
+
+  <dt>RFC pull request</dt>
+  <dd><a href="https://github.com/graphprotocol/rfcs/pull/11">URL</a></dd>
+
+  <dt>Obsoletes (if applicable)</dt>
+  <dd>-</dd>
+
+  <dt>Date of submission</dt>
+  <dd>2020-01-05</dd>
+
+  <dt>Date of approval</dt>
+  <dd>-</dd>
+
+  <dt>Approved by</dt>
+  <dd>-</dd>
+</dl>
+
+## Contents
+
+<!-- toc -->
+
+## Summary
+
+This RFC proposes adding a fulltext search filter type to the GraphQL API 
+to allow subgraph developers to specify language-specific, lexical, 
+composite filters that end users can use in their queries. The fulltext 
+index will be created by examining all words in a document, breaking it 
+into individul words and phrases (lexical analysis), and collapsing 
+variations of words into a single index term (stemming.)
+
+## Goals & Motivation
+
+The current set of string filters available in our GraphQL API is lacking 
+the tools needed to build effective, modern interfaces. Wildcard string 
+matching does provide string filtering, but users have come to expect the 
+easy to use filtering that comes with fulltext search systems.
+
+To facilitate building effective user interfaces we must provide human-user 
+friendly query filtering. Lexical, composite fulltext search filters can 
+provide the tools necessary for front-end developers to implement powerful 
+search bars that filter data across multiple fields of an Entity.
+
+With the proposed feature I aim to:
+  - provide tools for subgraph developers to define composite search 
+    indexes that search across multiple fields in an entity, and
+  - integrate powerful, user friendly text search that is flexible and 
+    can be easily defined by a subgraph developer ultimately providing 
+    the tools necesssary for front end developers to create effective 
+    user interfaces without the need to middleware.
+     
+## Urgency
+
+Fulltext search filters are not essential to the value proposition of 
```

Comment:
I don't see how it's true that fulltext search isn't essential to the value proposition of Graph Node. Disagree!

Reply:
Yep, I have no argument there; fulltext search style querying is vital to the value proposition.  

I believe what I was trying to convey is that: if the feature is not added immediately the Graph Node still has a strong value proposition and most current and prospective users will still be happy.  Maybe I'm being too literal about the difference between urgency and priority; but it's going to be a valuable feature that will likely get a lot of use, let's get it added right away! :)